src/duckling-tests/migration/existing-migration.spec.ts

```diff
@@ -0,0 +1,189 @@
+import { expect } from 'chai';
+import { Map } from 'immutable';
+
+import { MigrationService, ProjectVersionInfo } from '../../duckling/migration/migration.service';
+import { PathService } from '../../duckling/util/path.service';
+import { ChangeType, changeType } from '../../duckling/state/object-diff';
+import { ExistingCodeMigration } from '../../duckling/migration/existing-code-migration';
+import { MigrationTools } from '../../duckling/migration/migration-tools';
+import { ParsedMap } from '../../duckling/project/map-parser.service';
+import { EntitySystem, Entity, EntityKey } from '../../duckling/entitysystem/entity';
+
+
+const MIGRATION_FILE: ProjectVersionInfo = {
+    mapVersion: "1.0",
+    mapMigrations: [
+        {
+            type: "existing-code",
+            updateTo: "1.0",
+            name: "fake-collision-migration",
+            options: {
+                oldType: "oldType",
+                newType: "newType"
+            }
+        },
+    ]
+}
+
+const RAW_MAP: any = {
+    version: "0.0",
+    systems: {
+        collision: {
+            components: {
+                entity1: {
+                    collisionType: "oldType"
+                }
+            }
+        }
+    }
+}
+
+let fakeCollisionMigration: ExistingCodeMigration = {
+    rawMapFunction: (tools: MigrationTools) => {
+        return tools.attributeMigration("collision", (attribute: any, options: any) => {
+            if (attribute.collisionType === options.oldType) {
+                attribute.collisionType = options.newType;
+            }
+            return attribute;
+        });
+    },
+    entitySystemFunction: (entitySystem: EntitySystem, options?: any) => { 
+        return entitySystem.map((entity: Entity) => {
+            for (let attributeKey in entity) {
+                if (attributeKey === "collision") {
```

Comment:
Should use the COLLISION_KEY from "collision-attribute" since this is a game module test